### ISSUE TITLE
Declaration of Slim\Psr7\Uri::withScheme($scheme) must be compatible with Psr\Http\Message\UriInterface::withScheme(string $scheme): Psr\Http\Message\UriInterface 

### ISSUE THREAD

**Issue by george-slastnoy** [slimphp/Slim-Psr7#292]
Hello, after updating The psr/http-message to the latest version I have the following error:
Declaration of Slim\Psr7\Uri::withScheme($scheme) must be compatible with Psr\Http\Message\UriInterface::withScheme(string $scheme): Psr\Http\Message\UriInterface.
Will there be an update to Slim\PSR7?

**Comment by l0gicgate** [comment]
What version of PSR are you using?

**Comment by george-slastnoy** [comment]
Thanks for reply. How to check PSR version? 
I see now that the URI interface of the psr/http-message package is newer, the composer shows version 1.1, but I think it is 2.0 (withScheme method earlier did not use to contain the argument string type hint).

**Comment by george-slastnoy** [comment]
Also, there is psr/http-message version 2.0. We have 2 psr/http-message packages (one that came with a product we use, the other one came with a particular module of this application, but both of them are, I think, of April 2023).

**Comment by george-slastnoy** [comment]
I think the problem is that there are no type hints.

**Comment by george-slastnoy** [comment]
The active package is 2.0. Now the slim/psr7 package lacks function parameter and return hints.

**Comment by clownmeister** [comment]
Here is a workaround. Shouldn't really be used in production tho. I believe the pipeline is failing.

```
Hey so I started using `"slim/psr7": "dev-master#08dbc65964bf2c1fb0fe54256322a272cd02596b as 1.6.1",` as a workaround but only for some dev testing. 
We are kind of forced to stop using slim because of this. We will slowly make a transition to do so.

Note to the devs: Please try to release current master if possible
```
_Originally posted by @clownmeister in https://github.com/slimphp/Slim-Psr7/issues/296#issuecomment-2047947928_

**Comment by odan** [comment]
This have been fixed in v1.7.0.
